### DISCUSSION TITLE
No Slack notification method when using Report/Alert and Docker Compose

### DISCUSSION THREAD

**Discussion by ultrafab** [apache/superset#25487] [Q&A / Help]
Hello,
i'm trying to send a Chart Report to Slack, but during the configuration only "Email" method compares in the drop list.

I've inserted the Slack API value in the .env-non-dev file but nothing changes.
I'm using the non-dev docker compose with a dev worker, by the logs seems that the report is correctly generated, but i cannot complete the slack configuration in Superset.

Any hint?

Greetings,
Alessio

**Comment by ultrafab** [comment]
Found that i've to modify the superset_config.py and cannot pass those parameters in .end-no-dev

**Comment by benjamin-sicard-fipto** [comment]
Hello ser, what did you do exactly to solve your issue? Could you copy the content of our superset_config file?

Because I did everything according to the doc on the superset_config.py file, but I still cannot get Slack as a notification delivery method.
Thank you!

**Comment by Henri-J-Norden** [comment]
I had the same issue because I was trying to override the config with my file at `/app/superset_config.py`

But checking `echo $PYTHONPATH` I could see that /app/ was not included - and there is already a default docker config at `/app/docker/pythonpath_dev/superset_config.py`, which imports `superset_config_docker`. 
 
So **the solution** was to put my config at:

```
/app/docker/pythonpath_dev/superset_config_docker.py
```

_Maybe the documentation should be updated to more clearly reflect how overriding works in the docker-compose version?_

**Comment by hoon0912** [comment]
If SLACK_API_TOKEN conf value is empty, then this error happens.
please reference below link
https://github.com/apache/superset/blob/4.0.2/superset/views/base.py#L397

and you can find the information about SLACK_API_TOKEN from below link
https://superset.apache.org/docs/installation/kubernetes/#configuration-examples
